feat: Center Window Preview According to Current Dock Icon

chrisharper22 opened the pull request:
Calculate the window preview frame according to the hovered icon frame, attempting to replace #182 and update to modern codebase.

## Describe your changes
[SharedWindowPreviewCoordinator] - calculate x- and y-positions for the preview frame from the currently hovered Dock icon.
## Related issue number(s) and link(s)

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If this change affects core functionality, I have added a description highlighting the changes
- [x] I have followed conventional commit guidelines

## Core Functionality Changes
If this PR modifies core functionality, please provide a brief description of the changes and their impact below:
N/A

I am leaving this as a draft currently, as I do not have a reliable way to test for multi-screen setups. If someone is able to test this, I would appreciate it.

chrisharper22 commented:
Alright, so I've noticed an issue that occurs with multiple screens. When the main screen does not have the default screen resolution, the positioning of the preview will be calculated incorrectly on non-main screens. I will have to look into this.

ejbills commented on DockDoor/Views/Hover Window/SharedPreviewWindowCoordinator.swift:
```diff
@@ -162,20 +162,34 @@ final class SharedPreviewWindowCoordinator: NSWindow {
+        // Flip the coordinate space from the accessibility API (origin is bottom-left)
+        let flippedIconRect = CGRect(
+            x: iconRect.origin.x,
+            y: screenFrame.height - iconRect.origin.y - iconRect.height,
```
I have some helper methods called "cgRectFromNSRect" and vice versa which can do this for you.

chrisharper22 commented:
I updated the code to use the helper methods in `DockObserver` (e.g., `cgPointFromNSPoint(_:forScreen:)` to calculate the position for the appropriate coordinate space. I also updated the calculations for `yPosition` and `xPosition` due to this coordinate flipping (this needs to be double checked).

I tested by using my iPad and secondary MacBook to extend my screen using AirPlay, and from my testing the hover previews are in the correct locations per hovered icon, correct screens, correct distances from bounds/dock, _and_ correct screens.

@ejbills if you can test these changes, I would appreciate it.

ejbills commented:
@chrisharper22 this has been on the back-burner while I prioritized some other bugs. I will be taking a look at this this weekend! Apologies for the delay.

ejbills commented:
@chrisharper22 - apologies for the delay once again, I have finally found some time to test this thoroughly and I fixed a small issue. Besides that, it appears to be working as expected. Thank you for your continued additions!
